elastic/kibana
# Allow UserProfileSettingsClient to be created on Security Plugin Setup

## Summary

Closes https://github.com/elastic/kibana/issues/155949

The Security Plugin provides a UserProfileSettings client to the core UserSettingsService on Setup.

With the first implementation of Dark Mode, the client was not set until the security UserSettingService was started, which violates the setup/start contract best practices.

Now the client is provided during security plugin setup and the security UserSettingService will be set on start.

## Testing

Dark Mode should still work as it did before:

- Login as elastic
- Navigate to User Profile by clicking the Avatar menu > Edit Profile
- Choose Dark Mode, Save, and Refresh

- Bonus test: double check Adv. Settings Dark Mode vs User Profiles Dark Mode
